kivy/uix/textinput.py

```diff
@@ -835,7 +856,186 @@ def _set_undo_redo_bkspc(self, ol_index, new_index, substring, from_undo):
         #reset redo when undo is appended to
         self._redo = []
 
-    def do_cursor_movement(self, action):
+    def _move_word_left(self, index=None):
+        pos = index or self.cursor_index()
+        text = self.text
+        pos -= 1
+
+        if pos == 0:
+            return 0, 0
+
+        ucase = string.ascii_uppercase
+        lcase = string.ascii_lowercase
+        ws = string.whitespace
+        punct = string.punctuation
+
+        mode = 'normal'
+
+        c = text[pos]
+        if c in ws:
+            mode = 'ws'
+        elif c == '_':
+            mode = 'us'
+        elif c in punct:
+            mode = 'punct'
+        elif c not in ucase:
+            mode = 'camel'
+
+        while 0 < pos:
```

Comment:
I don't understand the need for going letter by letter for this, Can you explain why is this needed?